config.m4

```diff
@@ -24,19 +24,28 @@ if test "$PHP_MONGODB" != "no"; then
   dnl Check PHP version is compatible with this extension
   AC_MSG_CHECKING([PHP version])
 
-  if test -z "$PHP_CONFIG"; then
-    AC_MSG_ERROR([php-config not found])
-  fi
+  if test -z "${PHP_VERSION_ID}"; then
+    if test -z "$PHP_CONFIG"; then
+      AC_MSG_ERROR([php-config not found])
+    fi
 
-  PHP_MONGODB_PHP_VERSION=`${PHP_CONFIG} --version`
-  PHP_MONGODB_PHP_VERSION_ID=`${PHP_CONFIG} --vernum`
+    PHP_MONGODB_PHP_VERSION=`${PHP_CONFIG} --version`
+    PHP_MONGODB_PHP_VERSION_ID=`${PHP_CONFIG} --vernum`
+  else
+    PHP_MONGODB_PHP_VERSION="${PHP_VERSION}"
+    PHP_MONGODB_PHP_VERSION_ID="${PHP_VERSION_ID}"
+  fi
 
   AC_MSG_RESULT($PHP_MONGODB_PHP_VERSION)
 
   if test "$PHP_MONGODB_PHP_VERSION_ID" -lt "70400"; then
     AC_MSG_ERROR([not supported. Need a PHP version >= 7.4.0 (found $PHP_MONGODB_PHP_VERSION)])
   fi
 
+  PHP_MONGODB_STD_CFLAGS=""
+  PHP_MONGODB_DEV_CFLAGS=""
+  PHP_MONGODB_COVERAGE_CFLAGS=""
```

Comment:
Using prefixes _and_ initializing these vars ensures we won't inherit state during a static build (admittedly unlikely).